File: moody-study-backend/src/main/java/com/example/moody_study_backend/entity/SavedFile.java
package com.example.moody_study_backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "saved_files")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SavedFile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String fileName;
    private String fileType;

    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDateTime savedAt;
}

File: moody_study/lib/models/saved_file.dart
class SavedFile {
  final int id;
  final String fileName;
  final String fileType;
  final String content;
  final String savedAt;

  SavedFile({
    required this.id,
    required this.fileName,
    required this.fileType,
    required this.content,
    required this.savedAt,
  });

  factory SavedFile.fromJson(Map<String, dynamic> json) {
    return SavedFile(
      id: json['id'] as int,
      fileName: json['fileName'] as String,
      fileType: json['fileType'] as String,
      content: json['content'] as String,
      savedAt: json['savedAt'] as String,
    );
  }
}
